sklearn/ensemble/gradient_boosting.py

```diff
@@ -1417,11 +1420,21 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # init state
             self._init_state()
 
-            # fit initial model - FIXME make sample_weight optional
-            self.init_.fit(X, y, sample_weight)
+            # fit initial model
+            try:
+                self.init_.fit(X, y, sample_weight=sample_weight)
+            except TypeError:
+                if sample_weight_is_none:
+                    self.init_.fit(X, y)
+                else:
+                    raise ValueError(
+                        "The initial estimator {} does not support sample "
+                        "weights.".format(self.init_.__class__.__name__))
 
             # init predictions
-            y_pred = self.init_.predict(X)
+            y_pred = self.init_.predict(X).astype(np.float64, copy=False)
+            if y_pred.ndim == 1:
+                y_pred = y_pred[:, np.newaxis]
```

Comment:
```suggestion
                if self.loss_.K >= 2:
                    # multiclass classification needs y_pred to be of shape
                    # (n_samples, K) where K is the number of trees per
                    # iteration, i.e. the number of classes.
                    # see e.g. PriorProbabilityEstimator or ZeroEstimator
                    n_samples = y_pred.shape[0]
                    y_pred = np.tile(y_pred, self.n_classes_)
                    y_pred = y_pred.reshape(n_samples, self.loss_.K)
                else:
                    y_pred = y_pred[:, np.newaxis]
```

Reply:
Good catch. Wouldn't be better to make a one-hot of y_pred ?
I pushed it, let me know which version is best.

Reply:
Yes one-hot seems to be a better option. I think my version would have been equivalent to a zero initialization (or any constant value).

BTW I realized we could use ``if loss.is_multi_class`` to be more explicit.